src/common/eth-providers/consensus-provider/block-cache.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Inject, Injectable, LoggerService } from '@nestjs/common';
+import { BlockHeaderResponse, BlockInfoResponse } from './intefaces';
+import { LOGGER_PROVIDER } from '@lido-nestjs/logger';
+import { ConfigService } from 'common/config';
+import { Epoch, RootHex, Slot } from './types';
+
+export interface BlockCache {
+  missed: boolean;
+  header?: BlockHeaderResponse | void;
+  info?: BlockInfoResponse | void;
+}
+
+type BlockCacheId = Slot | RootHex;
+
+@Injectable()
+export class BlockCacheService {
+  constructor(@Inject(LOGGER_PROVIDER) protected readonly logger: LoggerService, protected readonly config: ConfigService) {}
+
+  private cache: { [slot: string]: BlockCache } = {};
+
+  public put(blockId: BlockCacheId, cache: BlockCache): void {
+    this.cache[String(blockId)] = cache;
+  }
+
+  public update(blockId: BlockCacheId, fresh: BlockCache): void {
+    // save only by slot number or root
+    if (['finalized', 'head'].includes(String(blockId))) return;
+    this.logger.debug(`Update ${blockId} ${Object.keys(fresh)} in slots cache`);
+    const old = this.get(String(blockId)) ?? {};
+    this.cache[String(blockId)] = { ...old, ...fresh };
+  }
```

Comment:
Maybe to remove?

Reply:
`put` is removed. Renamed `update` to `set`